x-pack/plugins/beats/server/lib/adapters/database/adapter_types.ts

```diff
@@ -0,0 +1,328 @@
+  docvalueFields?: DatabaseNameList;
+  fielddataFields?: DatabaseNameList;
+  from?: number;
+  ignoreUnavailable?: boolean;
+  allowNoIndices?: boolean;
+  expandWildcards?: ExpandWildcards;
+  lenient?: boolean;
+  lowercaseExpandedTerms?: boolean;
+  preference?: string;
+  q?: string;
+  routing?: DatabaseNameList;
+  scroll?: string;
+  searchType?: 'query_then_fetch' | 'dfs_query_then_fetch';
+  size?: number;
+  sort?: DatabaseNameList;
+  _source?: DatabaseNameList;
+  _sourceExclude?: DatabaseNameList;
+  _sourceInclude?: DatabaseNameList;
+  terminateAfter?: number;
+  stats?: DatabaseNameList;
+  suggestField?: string;
+  suggestMode?: 'missing' | 'popular' | 'always';
+  suggestSize?: number;
+  suggestText?: string;
+  timeout?: string;
+  trackScores?: boolean;
+  version?: boolean;
+  requestCache?: boolean;
+  index?: DatabaseNameList;
+  type?: DatabaseNameList;
+}
+
+export interface DatabaseSearchResponse<T> {
+  took: number;
+  timed_out: boolean;
+  _scroll_id?: string;
+  _shards: DatabaseShardsResponse;
+  hits: {
+    total: number;
+    max_score: number;
+    hits: Array<{
+      _index: string;
+      _type: string;
+      _id: string;
+      _score: number;
+      _source: T;
+      _version?: number;
+      _explanation?: DatabaseExplanation;
+      fields?: any;
+      highlight?: any;
+      inner_hits?: any;
+      sort?: string[];
+    }>;
+  };
+  aggregations?: any;
+}
+
+export interface DatabaseExplanation {
+  value: number;
+  description: string;
+  details: DatabaseExplanation[];
+}
+
+export interface DatabaseShardsResponse {
+  total: number;
+  successful: number;
+  failed: number;
+  skipped: number;
+}
+
+export interface DatabaseGetDocumentResponse<Source> {
+  _index: string;
+  _type: string;
+  _id: string;
+  _version: number;
+  found: boolean;
+  _source: Source;
+}
+
+export interface DatabaseBulkResponse {
+  took: number;
+  errors: boolean;
+  items: Array<
+    | DatabaseDeleteDocumentResponse
+    | DatabaseIndexDocumentResponse
+    | DatabaseUpdateDocumentResponse
+  >;
+}
+
+export interface DatabaseBulkIndexDocumentsParams
+  extends DatabaseGenericParams {
+  waitForActiveShards?: string;
+  refresh?: DatabaseRefresh;
+  routing?: string;
+  timeout?: string;
+  type?: string;
+  fields?: DatabaseNameList;
+  _source?: DatabaseNameList;
+  _sourceExclude?: DatabaseNameList;
+  _sourceInclude?: DatabaseNameList;
+  pipeline?: string;
+  index?: string;
+}
+
+export interface DatabaseMGetParams extends DatabaseGenericParams {
+  storedFields?: DatabaseNameList;
+  preference?: string;
+  realtime?: boolean;
+  refresh?: boolean;
+  _source?: DatabaseNameList;
+  _sourceExclude?: DatabaseNameList;
+  _sourceInclude?: DatabaseNameList;
+  index: string;
+  type?: string;
+}
+
+export interface DatabaseMGetResponse<T> {
+  docs?: Array<DatabaseGetResponse<T>>;
+}
+
+export interface DatabasePutTemplateParams extends DatabaseGenericParams {
+  id: string;
```

Comment:
One of the test failures I mentioned in the previous review comment originates from the fact that the param should probably be `name` instead of `id` (see https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html#api-indices-puttemplate).

Reply:
already fixed